### PR TITLE
Filter search by user or tag

### PR DESCRIPTION
**Pull request by oetherington**
Adds `user:name` and `tag:name` syntax to search queries. For both we accept a slug, name or id. Names with multiple words can be entered with quotes like `user:"jp addison"`.

Multiple filters of the same type are OR'd together. Multiple filters of different types are AND'd together.

After deploying, we need to run:
```
./scripts/serverShellCommand.sh "Globals.elasticDeleteIndex('Posts')"
./scripts/serverShellCommand.sh "Globals.elasticConfigureIndex('Posts')"
./scripts/serverShellCommand.sh "Globals.elasticExportCollection('Posts')"
```

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1208571899474226) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/search/SearchPageTabbed.tsx
Comment (oetherington) [inline packages/lesswrong/components/search/SearchPageTabbed.tsx:390]: This help message could probably be better.

**Comment by oetherington** [comment]
FYI @b0b3rt - this is not forum gated, and I think someone at LW requested this recently on Slack anyway. You'll need to run the commands from the PR description after deploy though.

File: packages/lesswrong/components/search/SearchPageTabbed.tsx
Comment (Will-Howard) [inline packages/lesswrong/components/search/SearchPageTabbed.tsx:390]:
```suggestion
            title={`"Quotes" and -minus signs are supported. Use user:"Jane Doe" or tag:"Expected value" to filter by user or tag.`}
```

**Review by Will-Howard — APPROVED** [review]
ASTLAC

1. Could you change all instances of "tag" to "topic" (or taggingNameSetting if possible)
2. It's a bit surprising that the name of the tag is highlighted in the snippet, not worth fixing if it's not immediately obvious though

Screenshot 2024-10-18 at 14 25 07

**Comment by oetherington** [comment]
@Will-Howard Thanks! I'll wait for @b0b3rt to approve before merging.

**Comment by b0b3rt** [comment]
Ack, sorry, will try to look at this ASAP.

**Review by b0b3rt — APPROVED** [review]
lgtm!

**Pull request merged**
